docs/core_tenets.md

```diff
@@ -0,0 +1,18 @@
+## Jujutsu's Core Tenets
+
+Jujutsu's core tenets are:
```

Comment:
Other design goals:

* Separation of logic and UI: It should be as easy to create new UIs (CLIs, GUIs, TUIs, servers) without having do duplicate logic. (We have a long way to go towards this goal.)
* Easy-to-use APIs: It should be easy to create new commands. For example, each command should not have to worry about concurrency, working-copy state, and rebasing descendants of rewritten commits.

Reply:
added